### DISCUSSION TITLE
When is 4.x planned for release?

### DISCUSSION THREAD

**Discussion by shahr** [dropwizard/dropwizard#6544] [Q&A]
Hi Dropwizard team,

The 4.x branch has been in beta for quite some time and I'm wondering when it is scheduled to be released?

Thank you for the efforts!

**Comment by sandeepk-veritas** [comment]
I also want to know more about same. Some guidance on tentative release date would help us in doing internal planning.

**Comment by joschi** [comment]
Unless any regressions are reported, Dropwizard [3.0.0-beta.5](https://github.com/dropwizard/dropwizard/discussions/6736) and [4.0.0-beta.4](https://github.com/dropwizard/dropwizard/discussions/6737) will be published as their respective GA release.

So give them a try and report anything not working for you. 😉

**Reply by sandeepnkulkarni** [reply]
Thanks for response, we are going to try 3.0.0 now. Any timeline in mind to have GA release? 

I can see there is a new project for junit4, https://github.com/dropwizard/dropwizard-testing-junit4. But it has no v3.0.0-beta.5 released, but instead v3.0.0-beta.4 was released. Currently we make use of ResourceTestRule, so it would really help us to keep using it to ease the migration. Earlier it was not available, so thanks for having it. Same request for v4.0.0-beta.4 on behalf of those planning to migrate to 4.0.0.

**Reply by joschi** [reply]
@sandeepnkulkarni You can use the `v3.0.0-beta.4` release of `dropwizard-testing-junit4` with Dropwizard 3.0.0-beta.5 for testing. 😄
